pkg/metrics/metrics.go

```diff
@@ -89,9 +89,7 @@ func initializeMetrics(m *MetricsConfig) (*MetricsConfig, error) {
 	return m, nil
 }
 
-func ShutdownController(pusher *controller.Controller, ctx context.Context, timeout time.Duration) {
-	ctx, cancel := context.WithTimeout(ctx, timeout)
-	defer cancel()
+func ShutDownController(pusher *controller.Controller, ctx context.Context) {
```

Comment:
By convention, the `context.Context` is the first parameter.

```suggestion
func ShutDownController(ctx context.Context, pusher *controller.Controller) {
```